qualtran/bloqs/mcmt/and_bloq.py

```diff
@@ -260,26 +256,38 @@ class MultiAnd(Bloq):
     Args:
         cvs: A tuple of control variable settings. Each entry specifies whether that
             control line is a "positive" control (`cv[i]=1`) or a "negative" control `0`.
+            If a (symbolic) integer is passed, assumes the control values to be all 1's.
```

Comment:
Is this subject to change / worth thinking about? I guess I haven't really thought about what symbolic controls mean.

Reply:
Its easy to argue for the case where the control spec is simply all 1s; and thats mostly the case I want to update everywhere we currently expect `cvs: Tuple[int, ...]` 

In future, we'd ideally like these bloqs to be able to take any `cvs: CtrlSpec` and it would be more nuanced to think about what a symbolic controlled spec would mean at that point. But for now, I just want to remove blockers for enabling symbolic resource estimation for higher level bloqs like state preparation via alias sampling / reflection etc.

Reply:
I'm not super thrilled about having one attribute named `cvs` that can represent _either_ a list of control values _or_ the number of controls. 

```python
def __init__(self, *, cvs=None, n_ctrl=None):
  if cvs is not Nont and n_ctrl is not None: raise ValueError()
  self._cvs = cvs
  self._n_ctrl = n_ctrl

  @property
  def cvs():
    ...
  @property
  def n_ctrl():
    ..
```

Reply:
Is there a way to do it without a custom init with frozendataclasses?  I can also do what I did for `QROM`; i.e. rename `cvs` -> `cvs_or_n_ctrl: Union[]` and then expose `cvs` and `n_ctrl` as properties.

Reply:
@mpharrigan  LMK if this approach sounds good and I'll make the changes.

Reply:
I think you can do this with attrs by prepending an underscore to the field name and setting `frozen(kwargs=True)`

Reply:
Then it would look something like this:

```
@frozen
class MultiAnd(Bloq):
    _cvs: Optional[Tuple[SymbolicInt, ...]] = field(
        converter=_to_tuple_or_none, kw_only=True, default=None
    )
    _n_ctrls: Optional[HasLength] = field(
        converter=_to_has_length_or_none, kw_only=True, default=None
    )

    def __attrs_post_init__(self):
        assert (self._cvs is None) ^ (self._n_ctrls is None)

```

We'll have to add this logic everywhere we use `cvs: Tuple[int, ...]` right now. I'm find doing this, but there's definitely more going on in this approach than having a single `Union` which must not be None.

Reply:
Also, the main reason for defining a new `Length` object is so that we can set `cvs` to be of type `Length` and `len(cvs)` still works. 

The above approach where we maintain both `n_ctrls` and `cvs` independently everywhere wouldn't require introducing `Length` since `n_ctrls` can just be `Symbolic`. 

At other places, we typically have a union of `Shaped` and numpy arrays so that `object.shape` works irrespective of the type of `object`. I think following a similar pattern here would keep things consistent. 

Tl;Dr - let's maintain `cvs: Union[HasLength, Tuple[int, ...]]`  or `cvs_or_n_ctrls: Union[HasLength, Tuple[int, ...]])` ? WDYT?

Reply:
oh oh oh when I first read it as `Union[Sequence, Length]` I was like, these are two different things. But `HasLength` ... yeah just one field named `cvs` is good

Reply:
can you update the docstring as well. instead of "if a symbolic integer is passed" say "if a HasLength object"